File: countMatches.py
from typing import List
import collections
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        ans = 0
        if not items:
            return 0

        for type_idx, color_idx, name_idx in items:
            if ruleKey == "type":
                if type_idx == ruleValue:
                    ans += 1

            if ruleKey == "color":
                if color_idx == ruleValue:
                    ans += 1

            if ruleKey == "name":
                if name_idx == ruleValue:
                    ans += 1

        return ans

File: C++_code/countMatches.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        unordered_map<string, int> info = {{"type",0}, {"color",1}, {"name",2}};
        int res = 0;
        int index = info[ruleKey];
        for (auto &&item : items) {
            if (item[index] == ruleValue) {
                res++;
            }
        }
        return res;
    }
};